lib/Db/LabelMapper.php

```diff
@@ -26,48 +26,113 @@
-		$sql = 'SELECT * FROM `*PREFIX*deck_labels` WHERE `board_id` = ? ORDER BY `id`';
-		return $this->findEntities($sql, [$boardId], $limit, $offset);
+	/**
+	 * @param numeric $boardId
+	 * @param int|null $limit
+	 * @param int|null $offset
+	 * @return Label[]
+	 * @throws \OCP\DB\Exception
+	 */
+	public function findAll($boardId, $limit = null, $offset = null): array {
+		$qb = $this->db->getQueryBuilder();
+		$qb->select('*')
+			->from($this->getTableName())
+			->where($qb->expr()->eq('board_id', $qb->createNamedParameter($boardId, IQueryBuilder::PARAM_INT)))
+			->setMaxResults($limit)
+			->setFirstResult($offset);
+		return $this->findEntities($qb);
 	}
 
-	public function delete(\OCP\AppFramework\Db\Entity $entity) {
+	/**
+	 * @param Entity $entity
+	 * @return Entity
+	 * @throws \OCP\DB\Exception
+	 */
+	public function delete(Entity $entity): Entity {
 		// delete assigned labels
 		$this->deleteLabelAssignments($entity->getId());
 		// delete label
 		return parent::delete($entity);
 	}
 
-	public function findAssignedLabelsForCard($cardId, $limit = null, $offset = null) {
-		$sql = 'SELECT l.*,card_id FROM `*PREFIX*deck_assigned_labels` as al INNER JOIN *PREFIX*deck_labels as l ON l.id = al.label_id WHERE `card_id` = ? ORDER BY l.id';
-		return $this->findEntities($sql, [$cardId], $limit, $offset);
+	/**
+	 * @param numeric $cardId
+	 * @param int|null $limit
+	 * @param int|null $offset
+	 * @return Label[]
+	 * @throws \OCP\DB\Exception
+	 */
+	public function findAssignedLabelsForCard($cardId, $limit = null, $offset = null): array {
+		$qb = $this->db->getQueryBuilder();
+		$qb->select('l.*', 'card_id')
+			->from($this->getTableName(), 'l')
+			->innerJoin('l', 'deck_assigned_labels', 'al', 'l.id = al.label_id')
+			->where($qb->expr()->eq('card_id', $qb->createNamedParameter($cardId, IQueryBuilder::PARAM_INT)))
+			->orderBy('l.id')
+			->setMaxResults($limit)
+			->setFirstResult($offset);
+
+		return $this->findEntities($qb);
 	}
-	public function findAssignedLabelsForBoard($boardId, $limit = null, $offset = null) {
-		$sql = 'SELECT c.id as card_id, l.id as id, l.title as title, l.color as color FROM `*PREFIX*deck_cards` as c ' .
-			' INNER JOIN `*PREFIX*deck_assigned_labels` as al ON al.card_id = c.id INNER JOIN `*PREFIX*deck_labels` as l ON  al.label_id = l.id WHERE board_id=? ORDER BY l.id';
-		return $this->findEntities($sql, [$boardId], $limit, $offset);
+
+	/**
+	 * @param numeric $boardId
+	 * @param int|null $limit
+	 * @param int|null $offset
+	 * @return Label[]
+	 * @throws \OCP\DB\Exception
+	 */
+	public function findAssignedLabelsForBoard($boardId, $limit = null, $offset = null): array {
+		$qb = $this->db->getQueryBuilder();
+		$qb->select('l.id as id', 'l.title as title', 'l.color as color')
+			->selectAlias('c.id', 'card_id')
+			->from($this->getTableName(), 'l')
+			->innerJoin('l', 'deck_assigned_labels', 'al', 'al.label_id = l.id')
+			->innerJoin('l', 'deck_cards', 'c', 'al.card_id = c.id')
```

Comment:
It seems there is a misspelling here 🙂. Instead it should be `->innerJoin('al', 'deck_cards', 'c', 'al.card_id = c.id')` I think.

Reply:
Maybe I misunderstood the phpDoc from `innerJoin` but my interpretation is that in the first argument you need to provide the alias for the `from` parameter.

> ```@param string $fromAlias The alias that points to a from clause.```